packages/evm/contracts/Hashi.sol

```diff
@@ -34,61 +34,89 @@
 pragma solidity ^0.8.17;
 
 import { IOracleAdapter } from "./interfaces/IOracleAdapter.sol";
+import { IHashi } from "./interfaces/IHashi.sol";
 
-contract Hashi {
-    error NoOracleAdaptersGiven(address emitter);
-    error OracleDidNotReport(address emitter, IOracleAdapter oracleAdapter);
-    error OraclesDisagree(address emitter, IOracleAdapter oracleOne, IOracleAdapter oracleTwo);
-
-    /// @dev Returns the hash reported by a given oracle for a given ID.
-    /// @param oracleAdapter Address of the oracle adapter to query.
-    /// @param domain Id of the domain to query.
-    /// @param id ID for which to return a hash.
-    /// @return hash Hash reported by the given oracle adapter for the given ID number.
-    function getHashFromOracle(
-        IOracleAdapter oracleAdapter,
+contract Hashi is IHashi {
+    /// @inheritdoc IHashi
+    function checkHashWithThresholdFromOracles(
         uint256 domain,
-        uint256 id
-    ) public view returns (bytes32 hash) {
-        hash = oracleAdapter.getHashFromOracle(domain, id);
+        uint256 id,
+        uint256 threshold,
+        IOracleAdapter[] calldata oracleAdapters
+    ) external view returns (bool) {
+        if (oracleAdapters.length < threshold) return false;
```

Comment:
given this would be an illegal call, shouldn't it revert rather than return false?

Reply:
makes sense!